### ISSUE TITLE
[auto-instrumentation] Init containers image run as root prevents usage of auto-injection where OPA Policies enforce runAsNonRoot

### ISSUE THREAD

**Issue by santhotech** [open-telemetry/opentelemetry-operator#2272]
### Component(s)

instrumentation

## Description

With the recent release of operators, the init-containers takes the securityContext of the container where they are injecting the instrumentation libraries . However, several enterprises enforce runAsNonRoot to be true in the application containers through OPA policies. With this update, when the initContainers inherit runAsNonRoot, they are erroring out with the following error
"Error: container has runAsNonRoot and image will run as root (pod: xxx, container: opentelemetry-auto-instrumentation-java)

## Steps to Reproduce

1. Created a pod with a container running a basic java application
2. Set the container securityContext to have runAsNonRoot: true
3. Create an instrumentation object 
4. Add the auto-instrumentation annotation to the container where instrumentation is required - instrumentation.opentelemetry.io/inject-java: true

## Expected Result

The initContainer should come up successfully and inject the instrumentation libraries

## Actual Result

The initContainer fails to come up with the error mentioned above.

### Kubernetes Version

1.23.0

### Operator version

0.87.0

### Collector version

0.87.0

**Comment by TylerHelmuth** [comment]
@santhotech I was not able to reproduce this problem locally yet.  When I set `runAsNonRoot: true` on the application container the view is correctly copied to the initContainer which runs as expected.

**Comment by santhotech** [comment]
@TylerHelmuth I noticed I have not mentioned, I am using auto-instrumentation-java image version 1.31.0, I was able to replicate it again. Can you clarify if this is the same version you tried?

**Comment by TylerHelmuth** [comment]
@santhotech are you able to reproduce this issue in a local kind cluster

**Comment by iblancasa** [comment]
With the current tests that @IshwarKanse implemented, I think we are covered to do this change... @patst would you be open to create a PR?

**Comment by patst** [comment]
> With the current tests that [@IshwarKanse](https://github.com/IshwarKanse) implemented, I think we are covered to do this change... [@patst](https://github.com/patst) would you be open to create a PR?

@iblancasa  yep sure enough. Created https://github.com/open-telemetry/opentelemetry-operator/pull/3986
